src/Provider/LinkedIn.php

```diff
@@ -161,7 +161,7 @@ protected function checkResponse(ResponseInterface $response, $data)
 
         if ($response->getStatusCode() >= 400) {
             throw new IdentityProviderException(
-                $data['message'] ?: $response->getReasonPhrase(),
+                isset($data['message']) ?: $response->getReasonPhrase(),
```

Comment:
@Addvilz I think this is effective at handling your use case where `$data['message']` is not set. I also think this will break the existing use case where `$data['message']` is set.

```
php > $data = [];
php > echo isset($data['message']) ?: 'default';
default
php > $data['message'] = 'Message!';
php > echo isset($data['message']) ?: 'default';
1
php >
```

You can see that if `$data['message']` is set the message that is provided to the `IdentityProviderException ` will be the boolean result from `isset($data['message'])`.

Reply:
I am also a bit concerned that the existing tests did not fail with this change!

Reply:
I've updated the test cases on master branch to properly check for not just the exception being thrown, but also the message and code of the exception. Master branch is now passing and this branch is not - due to the concern that I highlighted above :)

Reply:
Alright, I will update this branch and amend the fix.

Reply:
I uploaded an update with a change to fix this and a slight refactor to account for your changes in the other test. One thing: I used `::class`, is that okay?